File: api/src/main/java/org/egglog/api/notification/model/dto/response/NotificationResponse.java
package org.egglog.api.notification.model.dto.response;

import jakarta.persistence.*;
import lombok.*;
import org.egglog.api.notification.model.entity.enums.TopicEnum;
import org.egglog.api.user.model.entity.User;

import java.time.LocalDateTime;

/**
 * ```
 * ===================[Info]=========================
 * packageName    : org.egglog.api.notification.model.dto.response
 * fileName      : NotificationResponse
 * description    :
 * =================================================
 * ```
 * |DATE|AUTHOR|NOTE|
 * |:---:|:---:|:---:|
 * |2024-05-14|김형민|최초 생성|
 */
@Builder
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class NotificationResponse {

    private Long notificationId;

    private TopicEnum type;

    private boolean status;

    private LocalDateTime updatedAt;
}

File: client/data/src/main/java/com/org/egglog/data/setting/model/NotificationResponse.kt
package com.org.egglog.data.setting.model

import androidx.room.Entity
import androidx.room.PrimaryKey
import com.org.egglog.data.util.LocalDateTimeSerializer
import com.org.egglog.domain.setting.model.Notification
import com.org.egglog.domain.setting.model.NotificationType
import kotlinx.serialization.Serializable
import java.time.LocalDateTime

@Entity
@Serializable
data class NotificationResponse(
    @PrimaryKey val notificationId: Long,
    val type: NotificationType,
    val status: Boolean,
    @Serializable(with = LocalDateTimeSerializer::class) val updatedAt: LocalDateTime
)

fun NotificationResponse.toDomainModel(): Notification {
    return Notification(
        notificationId, type, status, updatedAt
    )
}